### PR TITLE
Update next-release.sh to make it work on Linux

### PR DESCRIPTION
**Pull request by exigow**
Changes in script:
* Choose between `gdate` and `date`
* Verify `gdate` or `date` is installed
* Verify `gh` is installed
* Update `cut` usage (not sure why, but -fX fields as separate arguments are unsupported on Linux)

### Test Plan

* Run `./next.release.sh` on macOS or Linux
* Expect output similar to `5.2.4100`

### CODE REVIEW

**Comment by exigow** [comment]
As a formality: I confirm that it works on Linux.

**Review by olafurpg — APPROVED** [review]
Just ran the script on my macOS to confirm it works correctly 👍🏻

**Comment by olafurpg** [comment]
BTW, this script will get replaced with a new `sg next-version` command. The tool `sg` can be installed in the sourcegraph/sourcegraph repository.

File: scripts/next-release.sh [hunk 1]
```diff
@@ -7,21 +7,38 @@
...
+    echo "Command not found: date"
+    exit 1
+  fi
+  date_program() {
+    date "$@"
+  }
 fi
 
+if ! command -v gh &>/dev/null; then
+  echo "Command not found: gh"
```
Comment (mkondratek) [inline scripts/next-release.sh:31]: would it make sense to add a similar message like for gdate? I mean smth like
```
    echo "The command gh is required to ..."
    echo "To fix this problem, run:\n  brew install gh"
```

**Comment by mkondratek** [comment]
I do confirm it works for me as expected! 🚀

**Review by mkondratek — APPROVED** [review]

File: scripts/next-release.sh (hunk 1 above)
Comment (exigow) [reply scripts/next-release.sh:31]: Naah, I'm just merging - looks like it is going to be removed soon. But thanks for comment!

**Pull request merged**